File: typescript/grains/grains.ts
export function square(num: number): bigint {
  const bigNum = BigInt(num);
  if (bigNum < 1n || bigNum > 64n) {
    throw new Error('Invalid input');
  }

  return 2n ** (bigNum - 1n);
}

export function total(): bigint {
  return 2n ** 64n;
}

File: go/grains/grains.go
package grains

import (
	"errors"
	"math"
)

// Square returns how many grains were on a given square
func Square(num int) (uint64, error) {
	if num < 1 || num > 64 {
		return 0, errors.New("Invalid input")
	}

	return uint64(math.Pow(2, float64(num-1))), nil
}

// Total returns the total number of grains on the chessboard
func Total() uint64 {
	return math.MaxUint64
}
